tests/em/static/test_DCIP_io_utils.py

```diff
@@ -0,0 +1,184 @@
+from __future__ import print_function
+
+# import matplotlib
+# matplotlib.use('Agg')
+import unittest
+import numpy as np
+
+from SimPEG.electromagnetics import resistivity as dc
+from SimPEG.electromagnetics.static import utils
+from SimPEG import maps, mkvc, data
+from SimPEG.utils.io_utils import io_utils_electromagnetics as io_utils
+import shutil
+import os
+
+
+class Test_DCIP2D_IO(unittest.TestCase):
+
+    def test_dc2d(self):
+        
+        # Survey parameters   
+        survey_type = ['pole-pole', 'dipole-dipole']
+        data_type = 'volt'
+        end_points = np.array([-100, 100])
+        topo = 10
+        num_rx_per_src = 4
+        station_spacing = 25
+
+        # Create sources and data object
+        source_list = []
+        for stype in survey_type:
+            source_list = source_list + utils.generate_dcip_sources_line(
+                stype, data_type, '2D', end_points, topo, num_rx_per_src, station_spacing
+            )
+        survey2D = dc.survey.Survey(source_list)
+        dobs = np.random.rand(survey2D.nD)
+        dunc = 1e-3*np.ones(survey2D.nD)
+        data2D = data.Data(survey2D, dobs=dobs, standard_deviation=dunc)
+
+        os.mkdir('./dcip2d_tests')
+
+        try:
+        # Write DCIP2D files
+            io_utils.write_dcip2d_ubc(
+                './dcip2d_tests/dcip2d_general.txt', data2D, 'volt', 'dobs', 'general', comment_lines="GENERAL FORMAT"
+            )
+            io_utils.write_dcip2d_ubc(
+                './dcip2d_tests/dcip2d_surface.txt', data2D, 'volt', 'dobs', 'surface', comment_lines="SURFACE FORMAT"
+            )
+            io_utils.write_dcip2d_ubc(
+                './dcip2d_tests/dcip2d_simple.txt', data2D, 'volt', 'dobs', 'simple', comment_lines="SIMPLE FORMAT"
+            )
+
+            # Read DCIP2D files
+            data_general = io_utils.read_dcip2d_ubc('./dcip2d_tests/dcip2d_general.txt', 'volt', 'general')
+            data_surface = io_utils.read_dcip2d_ubc('./dcip2d_tests/dcip2d_surface.txt', 'volt', 'surface')
+            data_simple = io_utils.read_dcip2d_ubc('./dcip2d_tests/dcip2d_simple.txt', 'volt', 'simple')
+
+            # Compare
+            passed = (np.all(
+               np.isclose(data2D.dobs, data_general.dobs) &
+               np.isclose(data2D.dobs, data_surface.dobs) &
+               np.isclose(data2D.dobs, data_simple.dobs)
+            ))
+
+            self.assertTrue(passed)
+            shutil.rmtree('./dcip2d_tests')
+            print("READ/WRITE METHODS FOR DC2D DATA PASSED!")
+
+        except Exception:
```

Comment:
`unittests` can have a `tearDown(self)` method that should be used to erase any generated files. I would recommend this over the try, except method you've used here so that if there is any error in the test it is slightly more clear where it came from. essentially you can put `shut.rmtree('../dcip2d_tests')` in that method.

Reply:
done